File: Data Structures & Algorithms/counting-bits/submission-1.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        # dp = [0] * (n + 1)
        # offset = 1
        # for i in range(1, n + 1):
        #     if offset * 2 == i:
        #         offset = i
        #     dp[i] = 1 + dp[i-offset]
        # return dp
        ans = []
        cnt = 0
        for i in range(n + 1):
            cnt = 0
            for j in range(32):
                if i & (1 << j):
                    cnt += 1
            ans.append(cnt)
        return ans

                

File: Data Structures & Algorithms/anagram-groups/submission-1.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> mp = new HashMap<>();
        for(String s : strs) {
            int[] cnt = new int[26];
            for(int i = 0; i < s.length(); i++) {
                cnt[s.charAt(i) - 'a']++;
            }

            String key = Arrays.toString(cnt);

            if(mp.containsKey(key)) {
                mp.get(key).add(s);
            }else {
                mp.put(key, new ArrayList<>());
                mp.get(key).add(s);
            }
        }
        List<List<String>> ans = new ArrayList<>();
        for(Map.Entry<String, List<String>> entry: mp.entrySet()) {
            ans.add(entry.getValue());
        }
        return ans;
    };
}
